### DISCUSSION TITLE
Migration from multi repo terraform code into a monolithic repository

### DISCUSSION THREAD

**Discussion by joeyqliu** [gruntwork-io/knowledge-base#763] [How do I]
Hey Terragrunt, 

I am looking to transform and migrate our current terraform setup into a monorepo with an end state that utilizes terragrunt. 
We currently have 20+ repositories mainly consisting of terraform code as stacks. I'm trying to pull all these stacks into a single repo and slowly refactor them to utilize terragrunt. 

Is there any guidance on doing something like this? 

Additionally, do you guys know if it is possible for me to have  a single repo, with a subdirectory containing stacks, but not using terragrunt & then to also have other subdirectories that do use terragrunt? 

I am trying to have a way for developers to slowly migrate over to the new structure while not blocking them on any other work. 

Thank you,
Joey

---

<ins datetime="2023-08-18T18:46:02Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110407">Tracked in ticket #110407</a></p>
</ins>

**Comment by lev-ok** [comment]
Hi @joeyqliu ,
 
Thank you for your questions!
 
> Is there any guidance on doing something like this?

I recommend you read the official Terragrunt documentation https://terragrunt.gruntwork.io/docs/
and this blog
https://blog.gruntwork.io/how-to-manage-multiple-environments-with-terraform-using-terragrunt-2c3e32fc60a8
 
 
> Additionally, do you guys know if it is possible for me to have a single repo, with a subdirectory containing stacks, but not using terragrunt & then to also have other subdirectories that do use terragrunt?

I assume yes, but you will most likely need to create a new sandbox account on AWS where you will test your Terragrunt configuration. 
 
Reagrds, 
Levko
